pkg/trace/api/api.go

```diff
@@ -762,3 +762,20 @@ func getMediaType(req *http.Request) string {
 	}
 	return mt
 }
+
+// setCommonHeaders enriches requests with headers setting custom container and additional tags and overwrites the
+// user agent header as well as sets the Via header.
+func setCommonHeaders(req *http.Request, tags string) {
```

Comment:
This is better, but we don't need the request anymore.
```suggestion
func setCommonHeaders(h http.Header, tags string) {
```